Tests/SwiftGRPCNIOTests/ServerThrowingTests.swift

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2018, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Dispatch
+import Foundation
+import NIO
+import NIOHTTP1
+import NIOHTTP2
+@testable import SwiftGRPCNIO
+import XCTest
+
+private let expectedError = GRPCStatus(code: .internalError, message: "expected error")
+
+// Motivation for two different providers: Throwing immediately causes the event observer future (in the
+// client-streaming and bidi-streaming cases) to throw immediately, _before_ the corresponding handler has even added
+// to the channel. We want to test that case as well as the one where we throw only _after_ the handler has been added
+// to the channel.
+private class ImmediateThrowingEchoProviderNIO: Echo_EchoProvider_NIO {
+  func get(request: Echo_EchoRequest, context: StatusOnlyCallContext) -> EventLoopFuture<Echo_EchoResponse> {
+    return context.eventLoop.newFailedFuture(error: expectedError)
+  }
+  
+  func expand(request: Echo_EchoRequest, context: StreamingResponseCallContext<Echo_EchoResponse>) -> EventLoopFuture<GRPCStatus> {
+    return context.eventLoop.newFailedFuture(error: expectedError)
+  }
+  
+  func collect(context: UnaryResponseCallContext<Echo_EchoResponse>) -> EventLoopFuture<(StreamEvent<Echo_EchoRequest>) -> Void> {
+    return context.eventLoop.newFailedFuture(error: expectedError)
+  }
+  
+  func update(context: StreamingResponseCallContext<Echo_EchoResponse>) -> EventLoopFuture<(StreamEvent<Echo_EchoRequest>) -> Void> {
+    return context.eventLoop.newFailedFuture(error: expectedError)
+  }
+}
+
+private extension EventLoop {
+  func newFailedFuture<T>(error: Error, delay: TimeInterval) -> EventLoopFuture<T> {
+    return self.scheduleTask(in: .nanoseconds(TimeAmount.Value(delay * 1000 * 1000 * 1000))) { () }.futureResult
```

Comment:
Why not use `seconds` here instead of multiplying?

Reply:
`TimeInterval` is floating-point, while `TimeAmount.Value` is integer-valued, so we'd lose a ton of precision.